### PR TITLE
bind_rows(.id = NULL) does not set names.

### LINKED ISSUE #5087: `bind_rows()` and row names in case of `.id = NULL`
In current development version if `.id` is not supplied (as by default) in `bind_rows()`, it forces new row names into output. I believe, this is consistent with underlying `vec_rbind()` (as it states [here](https://github.com/r-lib/vctrs/blob/5c69793052047bb5a395b6caadc00a2c32113843/R/bind.R#L35-L37)), but it is a surprise for 'dplyr' user.

I would also argue that creating row names in case of `.id = NULL` is a little bit excessive behavior.

Reprex of current situation:

``` r
library(dplyr, warn.conflicts = FALSE, quietly = TRUE)
packageVersion("dplyr")
#> [1] '0.8.99.9002'

bind_rows(list(a = tibble(x = 1:2))) %>% row.names()
#> [1] "a1" "a2"
```

<sup>Created on 2020-04-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by romainfrancois**
close #5087 I'll add a test if we decide to merge this

 ``` r
library(dplyr, warn.conflicts = FALSE, quietly = TRUE)
bind_rows(list(a = tibble(x = 1:2))) %>% row.names()
#> [1] "1" "2"
```

<sup>Created on 2020-04-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.3 (2020-02-29)
#>  os       macOS Mojave 10.14.6        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       Europe/Paris                
#>  date     2020-04-06                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version     date       lib source                        
#>  assertthat    0.2.1       2019-03-21 [1] CRAN (R 3.6.0)                
#>  backports     1.1.5       2019-10-02 [1] CRAN (R 3.6.0)                
#>  callr         3.4.3       2020-03-28 [1] CRAN (R 3.6.2)                
#>  cli           2.0.2       2020-02-28 [1] CRAN (R 3.6.0)                
#>  crayon        1.3.4       2017-09-16 [1] CRAN (R 3.6.0)                
#>  desc          1.2.0       2018-05-01 [1] CRAN (R 3.6.0)                
#>  devtools      2.2.2       2020-02-17 [1] CRAN (R 3.6.0)                
#>  digest        0.6.25      2020-02-23 [1] CRAN (R 3.6.0)                
#>  dplyr       * 0.8.99.9002 2020-04-06 [1] local                         
#>  ellipsis      0.3.0       2019-09-20 [1] CRAN (R 3.6.0)                
#>  evaluate      0.14        2019-05-28 [1] CRAN (R 3.6.0)                
#>  fansi         0.4.1       2020-01-08 [1] CRAN (R 3.6.0)                
#>  fs            1.4.0       2020-03-31 [1] CRAN (R 3.6.2)                
#>  generics      0.0.2       2018-11-29 [1] CRAN (R 3.6.0)                
#>  glue          1.4.0       2020-04-03 [1] CRAN (R 3.6.3)                
#>  highr         0.8         2019-03-20 [1] CRAN (R 3.6.0)                
#>  htmltools     0.4.0       2019-10-04 [1] CRAN (R 3.6.0)                
#>  knitr         1.28        2020-02-06 [1] CRAN (R 3.6.0)                
#>  lifecycle     0.2.0       2020-03-06 [1] CRAN (R 3.6.0)                
#>  magrittr      1.5         2014-11-22 [1] CRAN (R 3.6.0)                
#>  memoise       1.1.0       2017-04-21 [1] CRAN (R 3.6.0)                
#>  pillar        1.4.3       2019-12-20 [1] CRAN (R 3.6.0)                
#>  pkgbuild      1.0.6       2019-10-09 [1] CRAN (R 3.6.0)                
#>  pkgconfig     2.0.3       2019-09-22 [1] CRAN (R 3.6.0)                
#>  pkgload       1.0.2       2018-10-29 [1] CRAN (R 3.6.0)                
#>  prettyunits   1.1.1       2020-01-24 [1] CRAN (R 3.6.0)                
#>  processx      3.4.2       2020-02-09 [1] CRAN (R 3.6.0)                
#>  ps            1.3.2       2020-02-13 [1] CRAN (R 3.6.0)                
#>  purrr         0.3.3.9000  2020-03-31 [1] local                         
#>  R6            2.4.1       2019-11-12 [1] CRAN (R 3.6.0)                
#>  Rcpp          1.0.4.6     2020-04-03 [1] Github (RcppCore/Rcpp@2da7d9c)
#>  remotes       2.1.1       2020-02-15 [1] CRAN (R 3.6.0)                
#>  rlang         0.4.5.9000  2020-04-01 [1] Github (r-lib/rlang@a90b04b)  
#>  rmarkdown     2.1         2020-01-20 [1] CRAN (R 3.6.0)                
#>  rprojroot     1.3-2       2018-01-03 [1] CRAN (R 3.6.0)                
#>  sessioninfo   1.1.1       2018-11-05 [1] CRAN (R 3.6.0)                
#>  stringi       1.4.6       2020-02-17 [1] CRAN (R 3.6.0)                
#>  stringr       1.4.0       2019-02-10 [1] CRAN (R 3.6.0)                
#>  testthat      2.3.2       2020-03-02 [1] CRAN (R 3.6.0)                
#>  tibble        3.0.0       2020-03-30 [1] CRAN (R 3.6.3)                
#>  tidyselect    1.0.0       2020-01-27 [1] CRAN (R 3.6.0)                
#>  usethis       1.5.1.9000  2020-03-24 [1] Github (r-lib/usethis@01dbd8f)
#>  vctrs         0.2.99.9010 2020-04-06 [1] local                         
#>  withr         2.1.2       2018-03-15 [1] CRAN (R 3.6.0)                
#>  xfun          0.12        2020-01-13 [1] CRAN (R 3.6.0)                
#>  yaml          2.2.1       2020-02-01 [1] CRAN (R 3.6.0)                
#> 
#> [1] /Users/romainfrancois/.R/library/3.6
#> [2] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

</details>

### CODE REVIEW

File: R/bind.r [hunk 1]
```diff
@@ -116,7 +116,9 @@ bind_rows <- function(..., .id = NULL) {
   }
 
   dots <- map(dots, function(.x) if (is.data.frame(.x)) .x else tibble(!!!.x))
-
+  if (is.null(.id)) {
+    names(dots) <- NULL
```
Comment (hadley) [inline R/bind.r:120]: I don't think we need the conditional? To preserve existing dplyr behaviour, I think we could always strip the outer names.

Comment (romainfrancois) [reply R/bind.r:120]: tests disagree with that, e.g. 

``` r
library(dplyr, warn.conflicts = FALSE)
library(testthat)

test_that("bind_rows handles lists with NULL values (#2056)", {
  df1 <- tibble(x = 1, y = 1)
  df2 <- tibble(x = 2, y = 2)
  lst1 <- list(a = df1, NULL, b = df2)
  
  df3 <- tibble(
    names = c("a", "b"),
    x = c(1, 2),
    y = c(1, 2)
  )
  
  expect_identical(bind_rows(lst1, .id = "names"), df3)
})
#> Error: Test failed: 'bind_rows handles lists with NULL values (#2056)'
#> * <text>:15: bind_rows(lst1, .id = "names") not identical to `df3`.
#> Component "names": Modes: numeric, character
#> Component "names": target is numeric, current is character
```

<sup>Created on 2020-04-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Comment (hadley) [reply R/bind.r:120]: Oh doh, yeah, sorry, you're totally right.

Comment (romainfrancois) [reply R/bind.r:120]: Thanks. I'll add a test and merge then.

**Pull request merged**